### DISCUSSION TITLE
How to modify pyarmor 7 commands to pyarmor 8

### DISCUSSION THREAD

**Discussion by FuckingToasters** [dashingsoft/pyarmor#1699] [Q&A]
Hello, so i used following command in pyarmor 7.4 to obfuscate my scripts:

```
pyarmor pack -s Joinify.spec --clean -e " --add-binary 'c:\users\admin\appdata\local\programs\python\python39\lib\site-packages\tls_client\dependencies\tls-client-64.dll;tls_client\dependencies' --onefile --icon=icon.ico" -x " --enable-suffix --advanced 4 --restrict 102 --obf-mod 1 --obf-code 2" Joinify.py
```
Now i saw that pyarmor 8 seem to have more attributes for increased security like --assert-import or --enable-themida

Before i upgrade, i also want to use the existingf options from the old command provided above or if some of them got replaced with other, better ones, use them.  would be great if anyone can share some tips how i can modify the command to work with pyarmor 8.x

**Comment by jondy** [comment]
Some options in Pyarmor 7 are not available in Pyarmor 8.
In Pyarmor 8, first run pyinstaller to pack plain script
```
pyinstaller --add-binary ... --onefile --icon icon.ico Joinify.py
```
Then pass executable to `pyarmor gen`, for example,
```
pyarmor gen --pack dist/Joinify Joinify.py [other scripts or folder] 
```
Note that it need list all the scripts in the command lines.
Pyarmor 8 first obfuscate the scripts list in the command line, then replace the original one in the packed executable.
